esrally/mechanic/launcher.py

```diff
@@ -69,6 +69,8 @@ def start(self):
         for cluster_name, cluster_hosts in all_hosts.items():
             all_client_options = self.cfg.opts("client", "options").all_client_options
             cluster_client_options = all_client_options[cluster_name]
```

Comment:
You are modifying the object in the config directly which leads to side-effects, see e.g. https://github.com/elastic/rally/issues/472. Can you please make a copy with:

```python
cluster_client_options = dict(all_client_options[cluster_name])
```

Reply:
Great catch thanks!